ui/app/AppLayouts/Profile/helpers/SettingsEntriesModel.qml

```diff
@@ -0,0 +1,185 @@
+import QtQuick 2.15
+
+import StatusQ 0.1
+import utils 1.0
+
+import SortFilterProxyModel 0.2
+
+/*!
+    \qmltype SettingsEntriesModel
+    \inherits SortFilterProxyModel
+    \inqmlmodule AppLayouts.Profile.helpers
+
+    Model providing entries to the settings section.
+
+    Model structure:
+
+    subsection          [int]    - identifier of the entry (Constants.settingsSubsection)
+    text                [string] - readable name of the entry
+    icon                [string] - icon name
+    badgeCount          [int]    - number presented on the badge
+    isExperimental      [bool]   - indicates if the beta tag should be presented
+    experimentalTooltip [string] - tooltip text for the beta tag
+*/
+SortFilterProxyModel {
+    id: root
+
+    // Determines if wallet-related entries should be included
+    property bool showWalletEntries
+
+    // Determines if back up seed phrase entry should be included
+    property bool showBackUpSeed
+
+    // Badge count for the syncing entry
+    property int syncingBadgeCount: 0
+
+    // Badge count for the messaging section
+    property int messaginingBadgeCount: 0
+
+    readonly property string appsGroupTitle: qsTr("Apps")
+    readonly property string preferencesGroupTitle: qsTr("Preferences")
+    readonly property string aboutAndHelpGroupTitle: qsTr("About & Help")
+
+    readonly property var entries: [
+        {
+            subsection: Constants.settingsSubsection.backUpSeed,
+            text: qsTr("Back up seed phrase"),
+            icon: "seed-phrase"
+        },
+        {
+            subsection: Constants.settingsSubsection.profile,
+            text: qsTr("Profile"),
+            icon: "profile"
+        },
+        {
+            subsection: Constants.settingsSubsection.password,
+            text: qsTr("Password"),
+            icon: "password"
+        },
+        {
+            subsection: Constants.settingsSubsection.keycard,
+            text: qsTr("Keycard"),
+            icon: "keycard"
+        },
+        {
+            subsection: Constants.settingsSubsection.ensUsernames,
+            text: qsTr("ENS usernames"),
+            icon: "username",
+            isExperimental: true,
+            experimentalTooltip: qsTr("This section is going through a redesign.")
```

Comment:
I think you should define these 2 roles for other entries, even if they were empty

Reply:
it's not necessary, `ListModel` is smart enough to initialize roles properly when entries are provided in a one shot.